### DISCUSSION TITLE
How can I find the location where a snippet is coming from?

### DISCUSSION THREAD

**Discussion by ghost** [L3MON4D3/LuaSnip#942] [Q&A]
I'm inside a Rust project, and I get these 'Relm4' related snippets and I can't find where they're coming from. 

I'm very sure I haven't defined them anywhere, I don't have the package dowloaded anywhere. I don't understand where they might be defined.

<img width="2672" alt="image" src="https://github.com/L3MON4D3/LuaSnip/assets/75550932/e023d02d-4e40-4f2d-97d1-209cc9b0aabe">

I did manage to remove them by having the following in my `nvim-cmp` config:

```lua
      {
        name = "luasnip",
        priority = 4,
        entry_filter = function(entry, _)
          if string.match(entry.completion_item.label, "^relm%-") then return false end
          return true
        end,
      },
```

but I shouldn't be having these snippets appear in the first place

**Comment by L3MON4D3** [comment]
Oh, you should be able to find the location via [snip_location](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#snippet-location), as long as they're loaded by a loader.
If snip_location can't find it, the snippet is coming from lsp, and you probably can't do better than using some filter

**Reply by ghost** [reply]
I gave it a go with `jump_to_active_snippet` but it says the snippet is not associated with a loader. What's even more annoying that for example with Rust I have a snippet named "pfn", but the LSP also has one, and no matter what priority I set for mine, it's still not the first match. Is there a way to get around this?

The help is much appreciated!

**Reply by L3MON4D3** [reply]
Hm, yeah okay, then it's provided by LSP (I'm assuming you did set the `ls.setup`-variable which adds the source?)
It's possible to filter out specific results from lsp by intercepting the default callback (I've got something like that set up [here](https://github.com/L3MON4D3/Dotfiles/blob/505d92fe9092443f8ceeeaec58b08a8bede9b246/.config/nvim/lua/plugins/lspconfig.lua#L46-L63), but it isn't 100% your usecase, so you'd have to tinker with that a bit)

**Reply by ghost** [reply]
Thanks for that snippet of code! Although, in the end I've found the problem. I'm using AstroNvim which provides `friendly-snippets` and those relm related snippets and actually a bunch of others that I did not want we're coming from there. Disabling `friendly-snippets` solved the issue. Thanks for the help, I don't think I would've found their source if not for you!

**Reply by L3MON4D3** [reply]
Ah nice, glad I could help :D

**Comment by Lun4m** [comment]
Just bumped into this issue as well. These snippets come from the `frameworks` directory in friendly-snippets, which probably (at least the friendly-snippets [README.md](https://github.com/rafamadriz/friendly-snippets/blob/main/README.md#add-snippets-from-a-framework-to-a-filetype) suggests so) shouldn't be loaded by default in LuaSnip. 

I tried
```lua
require("luasnip.loaders.from_vscode").lazy_load({ exclude = { "frameworks" } })
```
but it doesn't do anything. And I can't figure out in which repo the problem is and if there is a solution :) 
Related #197 and rafamadriz/friendly-snippets#88 ?
